lib/kbuildbarn/urls.go

```diff
@@ -145,9 +68,12 @@ func readAndClose(rc io.ReadCloser) ([]byte, error) {
 
 // RetryUntilSuccess just blasts through all possible urls until it hits one that works. This is intended for
 // applications that are blind to the type of artifact
-func RetryUntilSuccess(params BuildBarnParams) ([]byte, error) {
+func RetryUntilSuccess(baseName, hash, size string) ([]byte, error) {
 	urls := []string{
-		params.FileUrl(), params.ActionUrl(), params.CommandUrl(), params.DirectoryUrl(),
+		Url(baseName, hash, size, WithActionUrlTemplate()),
+		Url(baseName, hash, size, WithDirectoryUrlTemplate()),
+		Url(baseName, hash, size, WithCommandUrlTemplate()),
+		Url(baseName, hash, size, WithByteStreamTemplate()),
```

Comment:
What about the file:// template? Should it be in this list too, or am I missing something?

Reply:
The File() endpoint ignores the scheme. If you wanted a file://<>/<>, you can use 
`kbuildbarn.WithFooTemplate(), kbuildbarn.WithScheme("file")`